cli/fathom_web/extract.py

```diff
@@ -0,0 +1,144 @@
+import base64
+import mimetypes
+import pathlib
+import shutil
+import re
+
+from click import argument, command, option, Path
+
+
+BASE64_DATA_PATTERN = re.compile(r'(data:(?P<mime>[a-zA-Z0-9]+?/[a-zA-Z0-9\-.+]+?);(\s?charset=utf-8;)?base64,(?P<string>[a-zA-Z0-9+/=]+))')
+BASE_TAG_PATTERN = re.compile(r'<base.*?>')
+OLD_CSP = re.compile(r"default-src 'none'; img-src data:; media-src data:; style-src data: 'unsafe-inline'; font-src data:; frame-src data:")
+NEW_CSP = r"default-src 'none'; img-src 'self' data:; media-src 'self' data:; style-src 'self' data: 'unsafe-inline'; font-src 'self' data:; frame-src 'self' data:"
+# These are MIME types the `mimetypes` library doesn't recognize or gets wrong. Matches were found at:
+# https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types
+MIME_TYPE_TO_FILE_EXTENSION = {
+    'application/font-woff': '.woff',
+    'application/font-woff2': '.woff2',
+    'application/vnd.ms-fontobject': '.eot',
+    'font/opentype': '.otf',
+    'font/ttf': '.ttf',
+    'font/woff2': '.woff2',
+    'image/jpeg': '.jpg',
+    'image/webp': '.webp',
+}
+
+
+@command()
+@option('--preserve-originals/--no-preserve-originals',
+        default=True,
+        help='Save original html files in a newly created `originals` directory in IN_DIRECTORY (default: True)')
+@argument('in_directory', type=Path(exists=True, file_okay=False))
+def main(in_directory, preserve_originals):
+    """
+    Extracts resources from the html pages in IN_DIRECTORY and stores them in a separate directory for Git-LFS storage.
+    """
+    if preserve_originals:
+        originals_dir = pathlib.Path(in_directory) / 'originals'
+        originals_dir.mkdir(parents=True, exist_ok=True)
+
+    for file in pathlib.Path(in_directory).iterdir():
+        if file.suffix != '.html':
+            print(f'Skipping {file.name}; not an html file')
```

Comment:
I get "Skipping originals; not an html file", which is kind of silly. :-)

Reply:
I'll put a separate check and message for directories.

Reply:
The weird part was just that fathom-extract had itself created that dir, so it was strange that it then turned around and complained about it.